Gems/ROS2/Code/Source/Lidar/ROS2LidarSensorComponent.cpp

```diff
@@ -154,31 +161,108 @@ namespace ROS2
         {
             point = inverseLidarTM.TransformPoint(point);
         }
-
+        typedef float PackedVector3[3];
```

Comment:
This is not the best approach - it may cause confusion, compared to standard types (like Vector3) used in O3DE. I would remove this typedef, and stay with floats.